File: backendApp/src/main/java/com/cinemas/entities/Room.java
package com.cinemas.entities;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "room")
@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "id")
public class Room {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column
    private String name;
    @Column
    private Integer SeatRows;
    @Column
    private Integer SeatColumns;
    @Column
    private Integer doubleSeatRows;
    @Column
    private Integer doubleSeatColumns;
    @Column
    private Integer totalColumn;

    @ManyToOne
    @JsonIgnore
    @JoinColumn(name = "cinema_id", referencedColumnName = "id")
    Cinema cinema;

    @OneToMany(mappedBy = "room", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Showtimes> showtimes = new ArrayList<>();

}

File: fllmbooking_app/lib/data/models/Room.dart
class Room {
  final int id;
  final String name;
  final int doubleSeatRows;
  final int doubleSeatColumns;
  final int totalColumn;
  final int seatColumns;
  final int seatRows;

  Room({
    required this.id,
    required this.name,
    required this.doubleSeatRows,
    required this.doubleSeatColumns,
    required this.totalColumn,
    required this.seatColumns,
    required this.seatRows,
  });

  factory Room.fromJson(Map<String, dynamic> json) {
    return Room(
      id: json['id'],
      name: json['name'],
      doubleSeatRows: json['doubleSeatRows'],
      doubleSeatColumns: json['doubleSeatColumns'],
      totalColumn: json['totalColumn'],
      seatColumns: json['seatColumns'],
      seatRows: json['seatRows'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'doubleSeatRows': doubleSeatRows,
      'doubleSeatColumns': doubleSeatColumns,
      'totalColumn': totalColumn,
      'seatColumns': seatColumns,
      'seatRows': seatRows,
    };
  }
}
